api/gateway/share.go

```diff
@@ -111,14 +111,17 @@ func (h *Handler) getShares(ctx context.Context, height uint64, nID namespace.ID
 	return shares.Flatten(), header.Height(), err
 }
 
-func dataFromShares(shares []share.Share) ([][]byte, error) {
-	blobs, err := appshares.ParseBlobs(shares)
```

Comment:
[for reviewers] this was the main issue. celestia-node was previously treating all types of shares as blob shares but now there are different types of shares (blobs occupy sparse shares and normal transactions / PFB transactions occupy compact shares). `ParseShares` is able to handle both types of shares.